src/puppeteer.js

```diff
@@ -150,6 +94,66 @@ const getPuppeteerOrThrow = (puppeteerModule = 'puppeteer') => {
     }
 };
 
+// TODO yin: `@property ...` wont work. It's may cause undefined property error in Typescript.
+//  It would be great if we can extend `LaunchOptions` from `puppeteer`. There is a GitHub issue for that:
+//  https://github.com/Microsoft/TypeScript/issues/20077
+/**
+ * Apify extends the launch options of Puppeteer.
+ * You can use any of the
+ * <a href="https://pptr.dev/#?product=Puppeteer&show=api-puppeteerlaunchoptions" target="_blank"><code>puppeteer.launch()</code></a>
+ * options in the [`Apify.launchPuppeteer()`](../api/apify#module_Apify.launchPuppeteer)
+ * function and in addition, all the options available below.
+ *
+ * @typedef {Object} LaunchPuppeteerOptions
+ * @property ...
```

Comment:
this is creating weird a declaration in `types\puppeteer.d.ts`, coming off as `(Missing): any`

Reply:
> the fix is to take the comment below the "..." and put on top of the typedef.

Can you provide a code sample? I feel stupid.

Reply:
``` 
 * You can use any of the
 * <a href="https://pptr.dev/#?product=Puppeteer&show=api-puppeteerlaunchoptions" target="_blank"><code>puppeteer.launch()</code></a>
 * options.
 *
 * @typedef {Object} LaunchPuppeteerOptions
 * @property {String} [proxyUrl]
 *   URL to a HTTP proxy server. It must define the port number,
 *   and it may also contain proxy username and password.
 *
```

the "@property ..." goes away (until Microsoft fix their tool)

Reply:
Microsoft won't. They use their open-source thralls for kind of things lately. We'll have to fix it ourselves.

One question with this: Isn't typescript going to scream whenever you use a puppeteer-original property? That should result in unknown property error, right?

Reply:
yes, it will, but that is the cast where you can just cast a `as PuppeteerOptions` after the object, as such:

```ts
// manually import the types from puppeteer
import { PuppeteerOptions } from 'puppeteer' 
import { LaunchPuppeteerOptions } from 'apify'

Apify.launchPuppeteer({ } as PuppeteerOptions & LaunchPuppeteerOptions) 
// should show all available options intersected
```

Reply:
Done. I added your snippet to the Typescript actor guide. Thanks!